unittests/savanna_misc_tests.cpp

```diff
@@ -1105,5 +1105,46 @@ BOOST_FIXTURE_TEST_CASE(finality_advancing_past_block_claimed_on_alternate_branc
 
 } FC_LOG_AND_RETHROW()
 
+// ------------------------------------------------------------------------------------
+// Test that replays blocks from fork_db at startup, and simulating a Ctrl-C
+// interruption of that replay.
+// (the cluster starts with 9 final blocks and 1 reversible block after the transition
+// to Savanna)
+// ------------------------------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE(replay_forkdb_at_startup, savanna_cluster::cluster_t) try {
+   auto& A=_nodes[0]; auto& C=_nodes[2]; auto& D=_nodes[3];
+
+   // at this point we have 9 final blocks and 1 reversible block
+
+   set_partition( { &C, &D } );                              // partition so blocks aren't finalized
+   const size_t num_blocks = 20;
+
+   std::vector<signed_block_ptr> blocks;
+   blocks.reserve(num_blocks);
+   for (size_t i=0; i<num_blocks; ++i)
+      blocks.push_back(A.produce_block());
+
+   const size_t num_forkdb_blocks = A.control->fork_db_size();;
+   BOOST_REQUIRE_GT(num_forkdb_blocks, num_blocks);        // A should have 20+ unfinalized blocks in its fork_db (actually 21)
+
+   controller::config copied_config = A.get_config();
+   auto               genesis       = block_log::extract_genesis_state(A.get_config().blocks_dir);
+
+   A.close();
+   A.remove_state();
+
+   A.open(make_protocol_feature_set(), genesis->compute_chain_id(), [genesis, &control=A.control]() {
+      auto check_shutdown = [](){
+         static size_t call_idx = 0;
+         return ++call_idx >= 15;                         // simulate Ctrl-C being hit on 15th call, so fewer than
+                                                          // 21 blocks from fork_db will be replayed.
+      };
+
+      control->startup([]() {}, check_shutdown, *genesis);
+   } );
+
+   BOOST_REQUIRE_EQUAL(A.control->fork_db_size(), num_forkdb_blocks);
+
```

Comment:
Why is this not 21 - 15?

Reply:
First, the blocks remain in the fork_db after being applied, because they are still not final.

Also `check_shutdown()` is also called when replaying the 9 blocks of the blocks log (so that's already 15 - 9), plus a couple times in between, so only 4 blocks from fork_db are replayed.

Reply:
Please fix the ${state} in remove state. 
Can you start it up after the stop during replay of forkdb to verify it restarts correctly after being interrupted during replay of forkdb.